documentation/markup.rst

```diff
@@ -10,6 +10,29 @@ This document describes the custom reStructuredText markup introduced by Sphinx
 to support Python documentation and how it should be used.
 
 
+Quick Reference
+===============
+
+This table summarizes which markup should be used for some commonly used
+elements:
+
+================ ========================================= ====================
+Element          Markup                                    See also
+================ ========================================= ====================
+arguments        ``*arg*``                                 :ref:`inline-markup`
+variables        ````var````                               :ref:`inline-markup`
+literals         ````0````, ````[]````                     :ref:`inline-markup`
+True/False/None  ````True````, ````False````, ````None```` :ref:`inline-markup`
+code snippets    ````print('hello world')````              :ref:`inline-markup`
+functions        ``:func:`sorted```                        :ref:`roles`
+func definitions ``.. function:: func(args)``              :ref:`directives`
+reference labels ``.. _label-name:``                       :ref:`doc-ref-role`
+internal ref     ``:ref:`label-name```                     :ref:`doc-ref-role`
+external links   ```Link text <https://target>`_``         :ref:`hyperlinks`
+comments         ``.. a comment``                          :ref:`comments`
+================ ========================================= ====================
```

Comment:
Some bikeshedding about the table:
* "Element" and "See also" could be replaced with other terms if you have better ideas.
* I tried to add an example of a string (` ````'string'```` `), but that doesn't seem to be possible without using ugly workarounds, so I just skipped it.
* I initially included more roles/directives (e.g. for methods/classes), but then decided to remove them after adding the "See also" column with the link to the relative sections.

Reply:
"Element" is good, IMO, and See Also is fine (though it *could* be replaced with something like "Reference" or "Section", I don't have a strong opinion).

IMO, I'd think it would be more useful to add methods and classes (and perhaps more) instead of repeating literals, variables _and_ code snippits (which my other suggestion consolidates into one)—those not only point to the same section, but actually are the exact same syntax, and it also makes the contrast with them and function parameters more clear, and the inclusion of `True`/`False`/`None` more purposeful (since otherwise, the reader would have no basis on which to think they would be formatted differently than other code/literals, and thus be confused at their explicit inclusion).

Reply:
> IMO, I'd think it would be more useful to add methods and classes (and perhaps more) 

The problem is that fitting `func`/`meth`/`cls` on one line is doable if we widen the table, but for the corresponding directive there's not enough space to add an example for each.  If we add all of them on separate lines, we will end up with a total of 6 total lines.  Listing all 3 together under the first column and only providing one example (e.g. using `:func:` and `.. function::` seems misleading.

Reply:
Oh, I originally meant adding functions, classes and methods as separate rows instead of variables, literals and code as separate rows. But I forgot about also adding references to them, which would mean 6 total rows which is a lot. So I agree with what you ended up doing :+1: